pyk/src/pyk/ktool/kprove.py

```diff
@@ -78,7 +79,7 @@ def _kprove(spec_file: str, *args: str) -> CompletedProcess:
 
 class KProve(KPrint):
 
-    def __init__(self, kompiled_directory, main_file_name=None, use_directory=None):
+    def __init__(self, kompiled_directory, main_file_name=None, use_directory=mkdtemp()):
         super(KProve, self).__init__(kompiled_directory)
         self.directory = Path(self.kompiled_directory).parent
         self.use_directory = (self.directory / 'kprove') if use_directory is None else Path(use_directory)
```

Comment:
I think there's a problem with using `mktemp()` as an default argument in this way.
It would reuse the same `mktemp()` directory for every instance of `KProve` we create. See [1].
This wouldn't be a big deal in thsi case because in most cases we would only create a single KProve instance in any program.

[1]: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Curious would following still work without the need of a private var and method for handling if default was None? 
If I understand init() should still get executed on each object instantiation of KProve. 
`        self.use_directory = (mkdtemp() / 'kprove') if use_directory is None else Path(use_directory)
        self.use_directory.mkdir(parents=True, exist_ok=True)
`
mkdtemp() will handle the folder creation process for us and pass back the full path. 
And to your point in 4. we can drop mkdir and require the user to create the directory. Could also provide a 'try/except' around line 84/85 to provide an error that folder must be provided ?

Reply:
> I'm not too familiar with all the options but specifically this problems had occurred from this call out of [evm-semantics ](https://github.com/runtimeverification/evm-semantics/blob/master/kevm_pyk/src/kevm_pyk/kevm.py#L94) and would test the default scenario. If there is a way to specifically run this change within that repository we can try that?

You can clone EVM semantics, and follow the instructions here to [build its dependencies.](https://github.com/runtimeverification/evm-semantics/blob/master/README.md#build-dependencies)
Make sure to update the KSubmodule before runing `make deps`.
The test that's failing is on [this line of the Jenkinsfile](https://github.com/runtimeverification/evm-semantics/blob/master/Jenkinsfile#L27).
That test would pass even without your PR right now, because the failure is caused by KEVM being packaged readonly.
This doesn't happen during the normal testing unfortunately, so lets manually make the EVM semantics kompiled directory (`find -name '*-kompiled'`) readonly before running the test to force the failure.

Reply:
> Curious would following still work without the need of a private var and method for handling if default was None?

`mkdtemp()` itself creates the temp directory. Its created whether or not we use `mkdir`. I'd rather not have that happen if we're not using it. We don't need the `kprove` subdir under it either.